design/one-pager-ignore-changes.md

```diff
@@ -0,0 +1,658 @@
+
+### ["Create", "Update", "Delete", "Observe", "LateInitialize"] or ["*"] by default
+
+Except for the cosmetic value, this decision affects how future additions to
+the management policy would be handled. If we go with the `["*"]` approach,
+every new management policy would be automatically included in the resources,
+which may be what we want. Similarly, if we go with the `["Create", "Update",
+"Delete", "Observe", "LateInitialize"]` approach, every new management policy
+would need to be added manually or through a migration.
+
+So this is something to keep in mind when developing new management policies.
+For this proposal, we will go with the `["*"]` approach as its more
+future-proof.
+
+### Migrating existing resources
+
+The `managementPolicy` feature is alpha, so it should be ok to change the
+API. We should still offer a migration path for existing resources. The
+combinations of `managementPolicy` and `deletionPolicy` should be migrated
+to the new `managementPolicy` field.
+
+| managementPolicy | deletionPolicy | new management policy                             |
+|------------------|----------------|---------------------------------------------------|
+| FullControl      | Delete         | ["*"]                                             |
+| FullControl      | Orphan         | ["Create", "Update", "Observe", "LateInitialize"] |
+| OrphanOnDelete   | Delete         | ["Create", "Update", "Observe", "LateInitialize"] |
+| OrphanOnDelete   | Orphan         | ["Create", "Update", "Observe", "LateInitialize"] |
+| ObserveOnly      | Delete         | ["Observe"]                                       |
+| ObserveOnly      | Orphan         | ["Observe"]                                       |
+
+### initProvider
+
+The idea here was to have a field that would be used only during creation,
+called `initProvider`. The `initProvider` field would have the same schema
+as `forProvider`. The `initProvider` fields would be merged with the
+`forProvider` fields during the `Create` step of the managed resource
+reconciliation loop. Fields that are specified in both `initProvider` and
+`forProvider` give precedence to the `forProvider` fields.
+
+```yaml
+apiVersion: eks.aws.crossplane.io/v1alpha1
+kind: NodeGroup
+metadata:
+  name: my-group
+spec:
+  initProvider:
+    scalingConfig:
+      desiredSize: 1
+  forProvider:
+    region: us-east-1
+    scalingConfig:
+      maxSize: 5
+      minSize: 1
+```
+
+This would allow users to specify fields that would only be used on creation,
+but not on updates.
+
+When it comes to the implementation, the handling would all need to be done
+on the provider level. The `initProvider` fields should only be used in the
+`Create` step of the managed resource reconciliation loop.
+
+Support for `initProvider` would need to be a manual change in each provider.
+From our side we could provide a helper function to merge the `initProvider`
+and `forProvider` fields which would be used in the `Create` step.
```

Comment:
Considering we are only aware a limited set of use cases/resources, we could also add support for `spec.initProvider` only for those resources. So, a managed resource may or may not have `spec.initProvider` depending on its nature. 

Another questions is if `initProvider` should have all the fields from `forProvider` or would it be ok just to put the fields that are needed, e.g. `scalingConfig.desiredSize` ?

Reply:
> Considering we are only aware a limited set of use cases/resources, we could also add support for `spec.initProvider` only for those resources. So, a managed resource may or may not have `spec.initProvider` depending on its nature.
> 

Added a paragraph below to better explain it. 

> We know that for now the `initProvider` field would be useful just for a 
handful of cases that we are aware of. So provider developers do not need to
support `initProvider` for all resources right away. Rather, they can add
support for those resources that are known to require it, and later on based
on a request basis.

So basically its up to the provider developers to support `spec.initProviders` for a particular MR, based on the needs. The important thing is that they have such an option now. 


> Another questions is if `initProvider` should have all the fields from `forProvider` or would it be ok just to put the fields that are needed, e.g. `scalingConfig.desiredSize` ?

I think it also depends on the provider developer. I suggested doing the same as `spec.forProvider` as it seems easier to maintain.